Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabUndo.cpp

```diff
@@ -33,12 +33,28 @@ namespace AzToolsFramework
         void PrefabUndoInstance::Capture(
             const PrefabDom& initialState,
             const PrefabDom& endState,
-            TemplateId templateId)
+            InstanceOptionalReference instanceRef)
```

Comment:
Seeking for suggestions: Is there a better way to avoid passing instance directly? It also requires casting out the const modifier which is not good IMO. My hunch is Link ID might work.